### DISCUSSION TITLE
Width's effect on χ

### DISCUSSION THREAD

**Discussion by qfan2** [Part-II-Computational-Physics/.github#61] [Q&A]
Hello,
Wonder if anyone else encountered this:
When I change the lattice width, the magnitude of χ changes a few orders of magnitude.

Thanks.

**Comment by PatrickC12** [comment]
I think this is expected behaviour. I observed Chi to change, but only by a factor of ~10 in going from 32 to 8 width.

maybe relevant: 

https://www.researchgate.net/figure/Variation-of-the-magnetic-susceptibility-and-of-the-magnetisation-in-the-insert-versus_fig2_225173987

**Comment by DLHvdH** [comment]
A change in chi is expected. Take a look at finite size scaling. Orders of magnitude might be a bit much though. There's a small mistake in the expression for the prefactor for the prefactor of the magnetic susceptibility. Make sure you only normalise the magnetisation by 1/N once and not twice.

**Reply by qfan2** [reply]
Thanks!
So just to double confirm:

$M = \frac{1}{N} \sum_i^N s_i$
(as usual)

$\chi= \frac{\partial M}{\partial H}=\beta\left(\left\langle M^2\right\rangle-\langle M \rangle^2\right)$
(Where I removed the division by N in χ)

And the expected behaviour is that the as the lattice size grows, χ grows (until the analytical solution corresponding to the infinite size lattice).

At the moment, I'm observing that every doubling in the lattice width corresponds to roughly a factor 10 *decrease* in χ (using the modified formulae above)

**Reply by DLHvdH** [reply]
That's correct!
